graphql-server/src/utils/cache.ts

```diff
@@ -0,0 +1,222 @@
+import fetch from "node-fetch";
+import NodeCache from "node-cache";
+import { props } from "./constants";
+import { neo4jDriver } from "./servers";
+import {
+  buildSamplesQueryBody,
+  buildSamplesQueryFinal,
+  queryDashboardSamples,
+} from "../schemas/queries/samples";
+import { DashboardRecordSort, DashboardSample } from "../generated/graphql";
+
+export const ONCOTREE_CACHE_KEY = "oncotree";
+export const SAMPLES_CACHE_KEY = "samples";
+
+const ONCOTREE_CACHE_TTL_SECONDS = 86400; // 1 day
+const SAMPLES_CACHE_TTL_SECONDS = 3600; // 1 hour
+
+const SAMPLES_PAGES_TO_CACHE = 5; // lower this if needed during dev for faster iterations
+const CACHE_BLOCK_SIZE = 100; // keep consistent with helpers.tsx's CACHE_BLOCK_SIZE
+// Keep consistent with SamplesList.tsx's DEFAULT_SORT
+const SAMPLES_DEFAULT_SORT = {
+  colId: "importDate",
+  sort: "desc",
+} as DashboardRecordSort;
+// Keep consistent with SamplesPage.tsx's WES_SAMPLE_CONTEXT
+const WES_SAMPLE_CONTEXT = {
+  fieldName: "genePanel",
+  values: [
+    "Agilent_51MB",
+    "Agilent_v4_51MB_Human",
+    "CustomCapture",
+    "IDT_Exome_v1_FP",
+    "IDT_Exome_V1_IMPACT468",
+    "WES_Human",
+    "WholeExomeSequencing",
+  ],
+};
+
+export type OncotreeCache = Record<string, { name: string; mainType: string }>; // key = Oncotree code
+export type SamplesCache = Record<string, DashboardSample[]>; // key = full Cypher query
```

Comment:
Current cache ≈ an object/hashmap where each key is an oncotree code and corresponding value contains the cancer type info

New cache ≈ 
```ts
{
    "oncotree": OncotreeCache,
    "samples": SamplesCache
}
```